presto-main/src/main/java/com/facebook/presto/sql/planner/LogicalPlanner.java

```diff
@@ -325,23 +338,78 @@ private RelationPlan createTableWriterPlan(
                     outputLayout));
         }
 
-        PlanNode writerNode = new TableWriterNode(
-                idAllocator.getNextId(),
-                source,
-                target,
-                symbols,
-                columnNames,
-                writerOutputs,
-                partitioningScheme);
+        List<Symbol> writerOutputs = ImmutableList.of(
+                symbolAllocator.newSymbol("partialrows", BIGINT),
+                symbolAllocator.newSymbol("fragment", VARBINARY));
+
+        List<Symbol> commitOutputs = ImmutableList.of(symbolAllocator.newSymbol("rows", BIGINT));
+
+        if (!statisticsMetadata.isEmpty()) {
+            verify(columnNames.size() == symbols.size(), "columnNames.size() != symbols.size(): %s != %s", columnNames.size(), symbols.size());
```

Comment:
nit: in case this fails some day, including full collections in the message might be helpful (perhaps even instead of the sizes):
```
verify(columnNames.size() == symbols.size(), "columnNames.size() != symbols.size(): %s and %s", columnNames, symbols);
```